src/core/ext/transport/chttp2/client/chttp2_connector.cc

```diff
@@ -56,7 +58,9 @@ void Chttp2Connector::Connect(const Args& args, Result* result,
   {
     MutexLock lock(&mu_);
     GPR_ASSERT(notify_ == nullptr);
-    args_ = args;
+    args_.channel_args = grpc_channel_args_copy(args.channel_args);
```

Comment:
Note that `Connect()` can be called multiple times on the same `Chttp2Connector` object, which would result in overwriting the previous value of `args_.channel_args` without freeing it -- i.e., a memory leak.

Instead of trying to solve this problem in the `Chttp2Connector` code specifically, I suggest the following:
- Change the `Connector::Connect()` API to take `args` by value instead of as a const reference, and have the subchannel code make a copy of the args before it passes them in.
- Add a dtor, move constructor, and move-assignment operator to the `Connector::Args` struct.
- Change the code here to just say `args_ = std::move(args);`.

Reply:
Thank you for the review! I will update ASAP!